### ISSUE TITLE
System.Diagnostics.Metrics.Tests.MetricEventSourceTests failing from event waiting timeouts

### ISSUE THREAD

**Issue by hoyosjs** [dotnet/runtime#56280]
Over the last 15 days, there's been a lot of outer loop failures for the metrics event source tests:

| Method | Type	| Count |
|:-- |:-- |:--|
| EventSourcePublishesEndEventsOnNewListener | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 236 |
| EventSourcePublishesTimeSeriesWithMetadata | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 236 |
| EventSourcePublishesTimeSeriesForLateInstruments | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 235 |
| EventSourcePublishesTimeSeriesWithTags | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 240 |
| EventSourcePublishesTimeSeriesWithEmptyMetadata | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 235 |
| EventSourcePublishesEndEventsOnMeterDispose | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 235 |
| EventSourcePublishesAllDataTypes | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 235 |
| EventSourceEnforcesTimeSeriesLimit | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 235 |
| EventSourceHandlesObservableCallbackException | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 235 |
| EventSourceEnforcesHistogramLimit | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 235 |
| EventSourcePublishesMissingDataPoints | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 235 |
| EventSourcePublishesInstruments | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 235 |
| EventSourcePublishesTimeSeriesForLateMeter | System.Diagnostics.Metrics.Tests.MetricEventSourceTests	| 235 |

See [this kusto query](https://dataexplorer.azure.com/clusters/engsrvprod/databases/engineeringdata?query=H4sIAAAAAAAAA2VSXW/TQBB8769YnSqaSLEvQaIIJFOJQkUQoJIW8YBQdbE38aa2z7pdJ43Ej2fPDmkrnuyd/ZrZOWth0TVCNcLlHFaOqi4gg29ASoTaURPxllqsqMGTz37JJ39gV2JA+N5hhwW8A7f2o9mrYnzM3KKrr4PfYC6QZWDabllRbsA1BSyw9Uziw75PFV4aFBsGEkPJje9Cjn06Jzs02+eFNuCKbYmuYBtZmie7WfhKhSixKfgAc8H6f+BWh/hOnmEfQ9BfRXRYe6D/zdU4AdU9LyagmloMQhiPsPF6nFFct0DuKmE4Uti3CCwuCO9ISjA3e9b56Qdy68azUM7pV5QQvz1dM44XP61wJWlcGKWfBlqXkurmiOg+fBDU4yixlrPWBca7Dftm9Ehq/Fj1hZr7jCXkTkZ9x6+zS19Vqoh88yPQ2e8JHPCDUREx9m7ZUVXodXtFF300LzJzrH4/IH31iy3hLqs53TInokKSHS7TvmcID2MScUvzhNwnfUwPkSFkcOBoSpGW31q1VHOpmp1Gt11LdqNvzpreg8EMY3c+3FN0LOI/NYj2XQXS4dX+X9GKKuQLnZBsMbDKzl5OZ2+S6Xkye23GfwGR5B0q+QIAAA==) for queries to the full builds and logs.

cc: @noahfalk @tarekgh

**Comment by tarekgh** [comment]
@noahfalk could you please have a look? Thanks!

**Comment by tarekgh** [comment]
@hoyosjs couldn't execute the query. do you have a better one? 

```
 'where' operator: Failed to resolve table or column expression named 'Jobs'

 clientRequestId: KustoWebV2;ecd650d0-fa01-4fa4-b660-ce9640bbb53e
```

**Comment by hoyosjs** [comment]
@tarekgh I opened the link again. Somehow it executes just fine. It needs to be done against the scope `@engsrvprod/engineeringdata`

**Comment by tarekgh** [comment]
Looks I am not authroized to execute it :-(

**Comment by hoyosjs** [comment]
[System.Diagnostics.Metrics.Tests.MetricEventSourceTests.xlsx](https://github.com/dotnet/runtime/files/6880281/export.5.xlsx)

**Comment by noahfalk** [comment]
It appears all the failures started 7/22 which was the exact day I checked in https://github.com/dotnet/runtime/pull/56083 and I forgot to run the outerloop tests. Decent odds this change broke it. Current hypothesis is that the tests are unexpectedly running in parallel and hitting the error handler that intentionally blocks concurrent usage of the EventSource. I'm working on a code change to the tests to see if we can confirm this.

**Comment by noahfalk** [comment]
I think this PR fixes it: https://github.com/dotnet/runtime/pull/56382

**Comment by hoyosjs** [comment]
Fixed in #56382
